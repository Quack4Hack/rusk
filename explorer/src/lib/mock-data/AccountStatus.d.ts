type AccountStatus = {
  balance: bigint;
  nonce: bigint;
};
