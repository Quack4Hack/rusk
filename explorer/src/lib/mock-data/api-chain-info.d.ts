type APIBlockHeader = {
  __typename: string;
  feespaid: number;
  hash: string;
  height: number;
  nextblockhash?: string;
  prevblockhash?: string;
  reward: number;
  seed: string;
  statehash?: string;
  timestamp: string;
  ts: number;
  version: string;
};

type APIBlockTransactions = {
  data: APITransaction[];
  stats: {
    averageGasPrice: number;
    gasLimit: number;
    gasUsed: number;
  };
};

type APIBlock = {
  __typename: string;
  header: APIBlockHeader;
  transactions: APIBlockTransactions;
};

type APIChainInfo = {
  blocks: APIBlock[];
  transactions: APITransaction[];
};

type APISearchBlockResult = {
  data: {
    data: {
      blocks: {
        header: { hash: string };
      }[];
    };
  };
};

type APISearchNoResult = {
  data: {};
};

type APISearchTransactionResult = {
  data: {
    data: {
      transactions: {
        __typename: string;
        tx: { id: string };
        txid: string;
      }[];
    };
  };
};

type APISearchResult =
  | APISearchBlockResult
  | APISearchNoResult
  | APISearchTransactionResult;

type APITransaction = {
  __typename: string;
  blockhash: string;
  blockheight: number;
  blocktimestamp: string;
  blockts: number;
  contract: string;
  feepaid: number;
  gaslimit: number;
  gasprice: number;
  gasspent: number;
  method?: string;
  success: boolean;
  txerror: string;
  txid: string;
  txtype: string;
};
