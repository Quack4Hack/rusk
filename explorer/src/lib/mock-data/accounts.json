[
  {
    "address": "0xe5601aae15c846ead69dab4c86a049405",
    "balance": 90993,
    "staked_balance": 13802,
    "active_stake": 9795,
    "inactive_stake": 4007,
    "rewards": 987
  },
  {
    "address": "0xea8bdd8e92be744e843c644bf581455ea",
    "balance": 83097,
    "staked_balance": 13179,
    "active_stake": 4469,
    "inactive_stake": 8710,
    "rewards": 3561
  },
  {
    "address": "0x883c4c22510df65fe29ebf6dd0ed710f6",
    "balance": 93610,
    "staked_balance": 49767,
    "active_stake": 39775,
    "inactive_stake": 9992,
    "rewards": 3616
  },
  {
    "address": "0x39aae05835f723cc8a7152b63f7d73ffc",
    "balance": 12804,
    "staked_balance": 5582,
    "active_stake": 4737,
    "inactive_stake": 845,
    "rewards": 4834
  },
  {
    "address": "0xad36bbf36e1cca4b3997abf14f611c2de",
    "balance": 92044,
    "staked_balance": 9446,
    "active_stake": 2944,
    "inactive_stake": 6502,
    "rewards": 1766
  },
  {
    "address": "0xdd211bb967327a5edd05592c5e44816e8",
    "balance": 84539,
    "staked_balance": 79912,
    "active_stake": 37637,
    "inactive_stake": 42275,
    "rewards": 2528
  },
  {
    "address": "0x1a229291051ec42a3e8abcfb7e9400cb0f",
    "balance": 61410,
    "staked_balance": 1460,
    "active_stake": 1324,
    "inactive_stake": 136,
    "rewards": 1399
  },
  {
    "address": "0xdc7f07a38332a73b6d3eb456c57548509",
    "balance": 35094,
    "staked_balance": 7814,
    "active_stake": 4616,
    "inactive_stake": 3198,
    "rewards": 972
  },
  {
    "address": "0x156a5e0592cfba9e2d6c9fe95d6f6bbbed",
    "balance": 85970,
    "staked_balance": 44119,
    "active_stake": 12865,
    "inactive_stake": 31254,
    "rewards": 4555
  },
  {
    "address": "0x127de00849726e14bb7e889f42fb1cc4d5",
    "balance": 32429,
    "staked_balance": 26072,
    "active_stake": 18899,
    "inactive_stake": 7173,
    "rewards": 849
  }
]
