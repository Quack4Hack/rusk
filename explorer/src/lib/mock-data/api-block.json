{
  "data": {
    "blocks": [
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 580718,
          "hash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
          "prevblockhash": "07b74b35c2c7cf8f41426cd0870bafa1a2c7adee3fdd876643548096186fc4cb",
          "nextblockhash": "6011556208a85e6001bd01ccbf936486b91318a7f6cbcf7ab810adf6fae34204",
          "reward": 16000000000,
          "seed": "af15447e3a004a79d4ae8b084f7b76b78d95880bb63e1cfa79250a310731f52e6d84ee42a5d6fc2cb99c5b1f489761f6",
          "statehash": "20bb0a677b93f084afadfd34bec3ac3feee33a020b81d9549afa2268e8543acb",
          "height": 495868,
          "ts": 1713298022,
          "timestamp": "2024-04-16 20:07:02 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [
            {
              "__typename": "SpentTransaction",
              "blockhash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
              "blockheight": 495868,
              "feepaid": 290866,
              "gasspent": 290866,
              "txerror": "",
              "success": true,
              "blockts": 1713298022,
              "blocktimestamp": "2024-04-16 20:07:02 +0000 UTC",
              "gaslimit": 500000000,
              "gasprice": 1,
              "txid": "3a3f6f90a1012ae751b4448bcb8e98def0ba2b18170239bd69fcf8e2e37f0602",
              "txtype": "1",
              "contract": "Transfer",
              "method": "transfer"
            },
            {
              "__typename": "SpentTransaction",
              "blockhash": "bd5c99bb720b03500e89f103fe66113ba62f2e124ed9651563f38fd15977719f",
              "blockheight": 495868,
              "feepaid": 289852,
              "gasspent": 289852,
              "txerror": "",
              "success": true,
              "blockts": 1713298022,
              "blocktimestamp": "2024-04-16 20:07:02 +0000 UTC",
              "gaslimit": 500000000,
              "gasprice": 1,
              "txid": "07bfabea1d94c16f2dc3697fa642f6cecea6e81bf76b9644efbb6e2723b76d00",
              "txtype": "1",
              "contract": "Transfer",
              "method": "transfer"
            }
          ],
          "stats": {
            "gasUsed": 580718,
            "averageGasPrice": 1,
            "gasLimit": 5000000000
          }
        }
      }
    ]
  }
}
