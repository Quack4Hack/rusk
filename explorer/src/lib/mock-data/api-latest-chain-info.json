{
  "data": {
    "blocks": [
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "8dc52b7a1d74df457397b3ddbdd0cc2ceb66d63f29f9336d68267f1d80960d57",
          "prevblockhash": "40d30b956fff04cf0305f6e47101edf482ff7bd1a9cc7f3ebb4e5dc7e7b50774",
          "reward": 16000000000,
          "seed": "929e2e8533c60461689183c7ca7425b847ada09446e48f209c7f2d21800dc2e9ffff3d8cf386446cd7cf32e5c71cc897",
          "height": 487541,
          "ts": 1713251354,
          "timestamp": "2024-04-16 07:09:14 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "40d30b956fff04cf0305f6e47101edf482ff7bd1a9cc7f3ebb4e5dc7e7b50774",
          "prevblockhash": "84c707df22777c8dda43f248d6c3107d98a2a4152d567621ec683ccc7cb49563",
          "reward": 16000000000,
          "seed": "a8dbb35b166a006e7ae0a7aecb693481b9804ae60220ed42caf909e1882db1bd534b343187decf0b1fc1205af1f932ba",
          "height": 487540,
          "ts": 1713251352,
          "timestamp": "2024-04-16 07:09:12 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "84c707df22777c8dda43f248d6c3107d98a2a4152d567621ec683ccc7cb49563",
          "prevblockhash": "8a63c6722da7ad1fb8c046a978452727b86f9dc4c2713c12d70867cedddcb774",
          "reward": 16000000000,
          "seed": "a5e471401f3d9e432eb8f386e66bc31ea5c46103a5dbdf45412781c45dec2c04f01ca3598bc452d5c9b4ba0b70b8fd83",
          "height": 487539,
          "ts": 1713251349,
          "timestamp": "2024-04-16 07:09:09 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "8a63c6722da7ad1fb8c046a978452727b86f9dc4c2713c12d70867cedddcb774",
          "prevblockhash": "aa003ef74fe0e7a53f4df32e54bdf28ffd5255601b7f4c00d9db839bde1477a6",
          "reward": 16000000000,
          "seed": "8e80c0275119eb24e6a3c1c7db9674d21210e2716de6ef5efc98aa9959219cc1e76f2aa3b1878e8215918257aeb7f83c",
          "height": 487538,
          "ts": 1713251346,
          "timestamp": "2024-04-16 07:09:06 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "aa003ef74fe0e7a53f4df32e54bdf28ffd5255601b7f4c00d9db839bde1477a6",
          "prevblockhash": "fa730b2417d729f790d8d5b43f4d9b39e3e9a1617f5926ae0f61ec0dd5b23196",
          "reward": 16000000000,
          "seed": "b2ee72056750dc5873c27d66637975bda303442dac3f3f640f465058acb3b0191100163a84e2926c294094956c7d69d6",
          "height": 487537,
          "ts": 1713251343,
          "timestamp": "2024-04-16 07:09:03 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "fa730b2417d729f790d8d5b43f4d9b39e3e9a1617f5926ae0f61ec0dd5b23196",
          "prevblockhash": "482d72696c605b79c78057a11d05187cd3575664914800696e9900be4a2deb82",
          "reward": 16000000000,
          "seed": "b761be817f2f6f941e8f958326db88bfd5209ccdab1db01ca71e61f80d783107d7c098cf9212f4534208c2da8829da69",
          "height": 487536,
          "ts": 1713251329,
          "timestamp": "2024-04-16 07:08:49 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "482d72696c605b79c78057a11d05187cd3575664914800696e9900be4a2deb82",
          "prevblockhash": "71b8d0cdc2d6e87fe7ece59c391ca69fe14c1240cd760bf1b406b8b4673edf4c",
          "reward": 16000000000,
          "seed": "b2b667c81e40107986fa89f68ced5524778ed06f8ec4b571467f74838fd1f5e1fa80d7aa2247daea5634851932c70169",
          "height": 487535,
          "ts": 1713251308,
          "timestamp": "2024-04-16 07:08:28 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "71b8d0cdc2d6e87fe7ece59c391ca69fe14c1240cd760bf1b406b8b4673edf4c",
          "prevblockhash": "886888f8460a3c00e8e1a3f8a30824d0c71e58fe794163e81308015c8e04ed7d",
          "reward": 16000000000,
          "seed": "a605c155d2e8c0e5c5b87d13013b2af1f06b354f312158aa4c1a4734ef892c45b5a92b0991e3733c34fff8dd3e5901be",
          "height": 487534,
          "ts": 1713251297,
          "timestamp": "2024-04-16 07:08:17 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "886888f8460a3c00e8e1a3f8a30824d0c71e58fe794163e81308015c8e04ed7d",
          "prevblockhash": "93853a6937ab8869378c03692472c088c5740596ca593344f005236de30cdc87",
          "reward": 16000000000,
          "seed": "b68fb78c947b85b50ad011f53e21dec284783d8abb0e65b02d1e0b165658f79edffd2e22b9cc546e533dce5a58a7a891",
          "height": 487533,
          "ts": 1713251294,
          "timestamp": "2024-04-16 07:08:14 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "93853a6937ab8869378c03692472c088c5740596ca593344f005236de30cdc87",
          "prevblockhash": "461de2ffa67923427b35d19485c35ebaecfffe5e7d97bde092cdfe20601fc5a2",
          "reward": 16000000000,
          "seed": "92e1188fdf8bcc72bc7a20c691d55db6c445c118ea601246776b24f34e2ac5d27edb73f7322c9f56bddcc7a9c52e21c4",
          "height": 487532,
          "ts": 1713251292,
          "timestamp": "2024-04-16 07:08:12 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "461de2ffa67923427b35d19485c35ebaecfffe5e7d97bde092cdfe20601fc5a2",
          "prevblockhash": "3fc46bdf145e4d0af3411e446cdfa0af9cdcf7cc61e6cf28025e33f31c9c97de",
          "reward": 16000000000,
          "seed": "93ef79b5fc4b27818af665bf2516f55773b96fc9bbb6e5e00921ac4df246ff57eaf985c2882272a8c11e9151d77e3eae",
          "height": 487531,
          "ts": 1713251288,
          "timestamp": "2024-04-16 07:08:08 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "3fc46bdf145e4d0af3411e446cdfa0af9cdcf7cc61e6cf28025e33f31c9c97de",
          "prevblockhash": "16b29f6868e979b4b19d3ed1e03f529775605c759a66d8ea071c3b7327664d17",
          "reward": 16000000000,
          "seed": "a00bd2a00d9fc1230a00757c8e029f161382f5073fde2670673501fbddcce1f8f7431b88278051685db71c53449bd56b",
          "height": 487530,
          "ts": 1713251286,
          "timestamp": "2024-04-16 07:08:06 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "16b29f6868e979b4b19d3ed1e03f529775605c759a66d8ea071c3b7327664d17",
          "prevblockhash": "004f07578fc2fe478e0b5b2cdc0b91568877d9c860b48c9a0ca07f44c1ad852e",
          "reward": 16000000000,
          "seed": "8b357b923f9dd9df93bba90432153d700d85840ea3c3fff0fd3d654696a4f67543896f4fc04594500e8d45667e52e850",
          "height": 487529,
          "ts": 1713251282,
          "timestamp": "2024-04-16 07:08:02 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "004f07578fc2fe478e0b5b2cdc0b91568877d9c860b48c9a0ca07f44c1ad852e",
          "prevblockhash": "0f0437797b4e3eac31458b760ab7b26b009756fe58f749c0dbf380a7a0493b8f",
          "reward": 16000000000,
          "seed": "9706007ea2bcf4a38434c5f73f20754caf0ad7ac4c1c6699807e3c72c82d54f3fd04df26ead778e90c7bafbc86fb04a1",
          "height": 487528,
          "ts": 1713251280,
          "timestamp": "2024-04-16 07:08:00 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      },
      {
        "__typename": "Block",
        "header": {
          "__typename": "Header",
          "feespaid": 0,
          "hash": "0f0437797b4e3eac31458b760ab7b26b009756fe58f749c0dbf380a7a0493b8f",
          "prevblockhash": "4b30046f32d0d6a0adae3bc20e8e75c6f6f7e4455e35388ca03e2e98eb69cbf2",
          "reward": 16000000000,
          "seed": "937256482180413da0c4e5c4a539c65c307d314936167c9b94c3dd30d30f79420aad66d907204c7618f3758916d1f985",
          "height": 487527,
          "ts": 1713251277,
          "timestamp": "2024-04-16 07:07:57 +0000 UTC",
          "version": "0"
        },
        "transactions": {
          "data": [],
          "stats": {
            "gasUsed": 0,
            "averageGasPrice": 0,
            "gasLimit": 5000000000
          }
        }
      }
    ],
    "transactions": [
      {
        "__typename": "SpentTransaction",
        "blockhash": "3c6e4018cfa86723e50644e33d3990bc27fc794f6b49fbf6290e4d308e07bd2d",
        "blockheight": 487166,
        "feepaid": 290766,
        "gasspent": 290766,
        "txerror": "",
        "success": true,
        "blockts": 1713249549,
        "blocktimestamp": "2024-04-16 06:39:09 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "4877687c2dbf154248d3ddee9ba0d81e3431f39056f82a46819da041d4ac0e04",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "40d3e59a9b2aeec95cc59e70fc61d387eb185f847d34aa033546c65e8ade4d3b",
        "blockheight": 487114,
        "feepaid": 290053,
        "gasspent": 290053,
        "txerror": "",
        "success": true,
        "blockts": 1713249234,
        "blocktimestamp": "2024-04-16 06:33:54 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "807d4f596ea083215bb38decf8b7e824e3efae734d6a43c2549717e57d04cf02",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "ea86c56ffa71ddd20a0268a7f1bb979b7adba54950b369597dcd6f8914074fe3",
        "blockheight": 487028,
        "feepaid": 290219,
        "gasspent": 290219,
        "txerror": "",
        "success": true,
        "blockts": 1713248832,
        "blocktimestamp": "2024-04-16 06:27:12 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "744b1d303f71e76e530b0543c284f65a30a26b75eab194d473c59d7fbd676a0b",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "ea86c56ffa71ddd20a0268a7f1bb979b7adba54950b369597dcd6f8914074fe3",
        "blockheight": 487028,
        "feepaid": 290460,
        "gasspent": 290460,
        "txerror": "",
        "success": true,
        "blockts": 1713248832,
        "blocktimestamp": "2024-04-16 06:27:12 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "34b4404adc0d286e62ddb9bf80429d4491b61de3f5e4c54975724a6d248d8e0b",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "a85a0f5cc265e156012cbb6dfd3652cd049201ae2c921175773ff9f0bb827312",
        "blockheight": 487022,
        "feepaid": 291263,
        "gasspent": 291263,
        "txerror": "",
        "success": true,
        "blockts": 1713248756,
        "blocktimestamp": "2024-04-16 06:25:56 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "88ac1818eb5aeee47a1356fe5d75726943c12ea1fb8ed23efb37c847a318ed0e",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "646bd535ae9e2d5ee021ed599f88ebf05f6a2a56fc3ccd4949db5767c1cb0906",
        "blockheight": 486999,
        "feepaid": 290735,
        "gasspent": 290735,
        "txerror": "",
        "success": true,
        "blockts": 1713248511,
        "blocktimestamp": "2024-04-16 06:21:51 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "3e430f7bbc123585a21b0689331d77a8c9e3da9b75ccc463aaf22d8a581e5507",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "fbcda951c660cdacaf3f1840de206c2fe6ca0cc5ef0647ec53cdc34bec844f78",
        "blockheight": 486954,
        "feepaid": 289627,
        "gasspent": 289627,
        "txerror": "",
        "success": true,
        "blockts": 1713248189,
        "blocktimestamp": "2024-04-16 06:16:29 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "776c0c54ccdd485fb32d6562b2b8b10f3cb283465da6eecba861f384725e3f0b",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "a90821aca4c51814af515d8b86cf655e2a54cf50e61e9f0b220c1a11467eb9c6",
        "blockheight": 486952,
        "feepaid": 290610,
        "gasspent": 290610,
        "txerror": "",
        "success": true,
        "blockts": 1713248160,
        "blocktimestamp": "2024-04-16 06:16:00 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "e022261204125d87ad99d404b73d0ca3ce4c0985e50f08b15a1619248ae0f702",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "4db9e793b1acba3f099c2927cf495caba3f9264c51bae24ba887d5e2ab163e74",
        "blockheight": 486950,
        "feepaid": 291347,
        "gasspent": 291347,
        "txerror": "",
        "success": true,
        "blockts": 1713248144,
        "blocktimestamp": "2024-04-16 06:15:44 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "fd9af8e161863389c2c52c900dc3881c754c7d2549f865df95ccfa0f57fe5b04",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "47c9f1349fd4894fc96b6e882fe0c23bc00cf600d26a466f1d22a4e7ce5103db",
        "blockheight": 486948,
        "feepaid": 290416,
        "gasspent": 290416,
        "txerror": "",
        "success": true,
        "blockts": 1713248120,
        "blocktimestamp": "2024-04-16 06:15:20 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "65be1885cbbf9545bc3b0f3c56b34da79672f2be063d81a5d4f78b2ca5c18101",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "4a62ff5839afd5127f99953a5b65ba2221f67239cce8dbe391a4f66ad8084dca",
        "blockheight": 486446,
        "feepaid": 290006,
        "gasspent": 290006,
        "txerror": "",
        "success": true,
        "blockts": 1713245391,
        "blocktimestamp": "2024-04-16 05:29:51 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "80f99e41794be445401bff0d8502b093fa41a88a6c3ed63061a7ec8d3fdaf30c",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "ce994d006c9143eb6d980e1aa74097252c90293e40bd22d93d222d966e9c1499",
        "blockheight": 486445,
        "feepaid": 290424,
        "gasspent": 290424,
        "txerror": "",
        "success": true,
        "blockts": 1713245384,
        "blocktimestamp": "2024-04-16 05:29:44 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "1391f6681e1437f4aec9db9de0d353aca41000e24bd2e4e465d4ec54c1ce8502",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "e852377c096a59c0735536996e842622373ce927325daccae6ffb49e20371c77",
        "blockheight": 486416,
        "feepaid": 290304,
        "gasspent": 290304,
        "txerror": "",
        "success": true,
        "blockts": 1713245224,
        "blocktimestamp": "2024-04-16 05:27:04 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "9cc2ad9734df30b12e0283b5a40a710e334351399da4225618cb65c7982e1d04",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "d3d49fff7bbd38041935c86c452b3490e5c25f1ba9218ad5268770e944085bb8",
        "blockheight": 486352,
        "feepaid": 290815,
        "gasspent": 290815,
        "txerror": "",
        "success": true,
        "blockts": 1713244833,
        "blocktimestamp": "2024-04-16 05:20:33 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "63c782c0bf3867ecac8f58a58902d264ccb0c4aa40af16dbc1223e0e18d5c902",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      },
      {
        "__typename": "SpentTransaction",
        "blockhash": "a3f465cce540c62eb28eafc8ffb68346839c84b133a5fd45f0daec82162bd94f",
        "blockheight": 485983,
        "feepaid": 289437,
        "gasspent": 289437,
        "txerror": "",
        "success": true,
        "blockts": 1713242593,
        "blocktimestamp": "2024-04-16 04:43:13 +0000 UTC",
        "gaslimit": 500000000,
        "gasprice": 1,
        "txid": "052134003432470e86e62d127b97d28b2541110a78f0c79f4407ee778afe5e04",
        "txtype": "1",
        "contract": "Transfer",
        "method": "transfer"
      }
    ]
  }
}
