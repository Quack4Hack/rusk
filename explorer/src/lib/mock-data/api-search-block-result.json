{
  "data": {
    "data": {
      "blocks": [
        {
          "header": {
            "hash": "fda46b4e06cc78542db9c780adbaee83a27fdf917de653e8ac34294cf924dd63"
          }
        }
      ]
    }
  }
}
