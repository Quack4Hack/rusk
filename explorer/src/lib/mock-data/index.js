export { default as apiMarketData } from "./api-market-data.json";
export { default as apiNodeLocations } from "./api-node-locations.json";
export { default as apiStats } from "./api-stats.json";
export { default as gqlBlock } from "./gql-block.json";
export { default as gqlBlocks } from "./gql-blocks.json";
export { default as gqlLatestChainInfo } from "./gql-latest-chain-info.json";
export { default as gqlSearchPossibleResults } from "./gql-search-possible-results.json";
export { default as gqlTransaction } from "./gql-transaction.json";
export { default as gqlTransactionDetails } from "./gql-transaction-details.json";
export { default as gqlTransactions } from "./gql-transactions.json";
