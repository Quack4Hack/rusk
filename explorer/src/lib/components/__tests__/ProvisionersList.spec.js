import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { slice } from "lamb";

import { hostProvisioners } from "$lib/mock-data";

import { ProvisionersList } from "..";

describe("Provisioners List", () => {
  const provisioners = slice(hostProvisioners, 0, 1)[0];

  /** @type {import("svelte").ComponentProps<ProvisionersList>} */
  const baseProps = { data: provisioners };

  afterEach(cleanup);

  it("renders the `ProvisionersList` component", () => {
    const { container } = render(ProvisionersList, baseProps);

    expect(container.firstChild).toMatchSnapshot();
  });
});
