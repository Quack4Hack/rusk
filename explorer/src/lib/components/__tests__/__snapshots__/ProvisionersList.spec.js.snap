// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Provisioners List > renders the \`ProvisionersList\` component 1`] = `
<dl
  class="detail-list provisioners-list"
>
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      Staking address
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    <span
      class="provisioners-list__staking-address"
    >
      m6dy2gz3jC3ifLpZmdTaZbgcp...uSCne1ti5kgYeNHtW82FWtnpf
    </span>
  </dd>
   
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      Owner
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    <span
      class="dusk-badge dusk-badge--variant-neutral"
      data-tooltip-id="provisioners-tooltip"
      data-tooltip-text="m6dy2gz3jC3ifLpZmdTaZbgcp...uSCne1ti5kgYeNHtW82FWtnpf"
    >
      Account
    </span>
  </dd>
   
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      Active Stake
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    500,000
     DUSK
  </dd>
   
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      Inactive Stake
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    0
     DUSK
  </dd>
   
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      Maturity At
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    #
    0
  </dd>
   
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      Soft Slashes
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    0
  </dd>
   
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      Hard Slashes
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    0
  </dd>
   
  <dt
    class="details-list__term details-list__term--table"
  >
    <div
      class="details-list__term-layout"
    >
       
      accumulated reward
    </div>
  </dt>
   
  <dd
    class="details-list__definition details-list__definition--table"
  >
    7,121.257296087
     DUSK
  </dd>
</dl>
`;
